netlify/functions/feedback-management.js

```diff
@@ -0,0 +1,130 @@
+const { WebClient } = require('@slack/web-api');
+const querystring = require('node:querystring');
+const { JWT } = require('google-auth-library');
+const { sheets } = require('@googleapis/sheets');
+
+const slackChannelIds = {
+  'mui-x': 'C04U3R2V9UK',
+  'mui-core': 'C041SDSF32L',
+};
+
+const spreadSheetsIds = {
+  forLater: '1NAUTsIcReVylWPby5K0omXWZpgjd9bjxE8V2J-dwPyc',
+};
+
+// Slack API
+const slackClient = new WebClient(process.env.SLACK_BOT_TOKEN);
+
+/**
+ * @param {object} event
+ * @param {object} context
+ */
+exports.handler = async (event) => {
+  try {
+    const { payload } = querystring.parse(event.body);
+    const data = JSON.parse(decodeURIComponent(payload));
+
+    // eslint-disable-next-line no-console
+    console.log('=== data ===');
+    // eslint-disable-next-line no-console
+    console.log(JSON.stringify(data, null, 2));
+    // eslint-disable-next-line no-console
+    console.log('============');
+    switch (data.callback_id) {
+      case 'send_feedback':
+        {
+          // We send the feedback to the appopiate slack channel
+          const { rating, comment, currentLocationURL, commmentSectionURL, commmentSectionTitle } =
+            data;
+
+          const simpleSlackMessage = [
+            `New comment ${rating === 1 ? '👍' : ''}${rating === 0 ? '👎' : ''}`,
+            `>${comment.split('\n').join('\n>')}`,
+            `sent from ${currentLocationURL}${
+              commmentSectionTitle
+                ? ` (from section <${commmentSectionURL}|${commmentSectionTitle})>`
+                : ''
+            }`,
+          ].join('\n\n');
+
+          const isDocsX = currentLocationURL.includes('/x/');
+          await slackClient.chat.postMessage({
+            channel: slackChannelIds[isDocsX ? 'mui-x' : 'mui-core'],
+            text: simpleSlackMessage,
+            as_user: true,
+          });
+        }
+        break;
+      case 'delete_action':
+        await slackClient.chat.delete({
```

Comment:
Wouldn't this allow random people to delete slack messages in our org?

Perhaps this needs some form of authentication?

Reply:
If they get access to the `message_ts` yes. Effectively right be a good point to add authentication of the request origine

Reply:
We could configure a simple API key?